### ISSUE TITLE
Wrong computation of K.batch_dot with negative index in axes

### ISSUE THREAD

**Issue by lucagrementieri** [keras-team/keras#8690]
Running the following code I was expecting to get an error because the summation is performed over axis of different size, but it is not the case.
Instead the output is `(1, 2, 3, 5, 6)`, meaning that the second-last dimension is used.
```
import keras.backend as K

t = K.random_uniform_variable((1,2,3,4,5),0,1)
r = K.random_uniform_variable((1,2,3,4,6),0,1)
p = K.batch_dot(t,r, axes=-1)
result = K.get_session().run(p)
print(result.shape)
```
More in general I have observed that if `axes` is a negative number the summation is always performed over the second-last dimension.
